src/schema/types.rs

```diff
@@ -79,6 +80,51 @@ impl Number for i128 {}
 impl Number for f32 {}
 impl Number for f64 {}
 
+#[derive(Clone, Debug, Display, PartialEq, Eq)]
+#[display(Debug)]
+/// Error indicating that a value is not in 0..=18 and therefore not suitable for a [DecimalPrecision]
+pub struct DecimalPrecisionOverflowError;
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Display)]
```

Comment:
Without `Hash` you can't use it as a keys in `BTreeMap`

```suggestion
#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug, Display)]
```